src/lib.rs

```diff
@@ -2964,6 +2976,17 @@ impl Connection {
             }
         }
 
+        // Create PADDING frame for the rest of the packet.
+        // PADDING frame can't be a single frame in the packet,
+        // so only add it when there is a frame already.
+        if add_padding && left >= 1 && !frames.is_empty() {
```

Comment:
Should probably move this a few lines down after the existing `frames.is_empty()` check, so we don't have to duplicate it.

Reply:
Moved.